### PR TITLE
scaffold fsm supports pre/post db conditions

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

scaffold fsm supports pre/post db conditions

## Types of changes

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [x] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/fsm/scaffold/generators/components.py [hunk 1]
```diff
@@ -70,11 +70,25 @@ def get_file_content(self) -> str:
             COPYRIGHT_HEADER,
             ROUNDS.HEADER.format(**self.template_kwargs),
             self._get_rounds_section(),
-            self.ABCI_APP_CLS.format(**self.template_kwargs),
+            self.ABCI_APP_CLS.format(
+                **self.template_kwargs,
+                db_pre_conditions=self._get_pre_condition_rounds_str(),
+                db_post_conditions=self._get_post_condition_rounds_str(),
+            ),
```
Comment (Adamantios) [inline autonomy/fsm/scaffold/generators/components.py:77]: Let's do this in the template instead:
https://github.com/valory-xyz/open-autonomy/blob/f4f5e704c2eee9e21deb6c7245282f8850554348/autonomy/fsm/scaffold/base.py#L119

File: autonomy/fsm/scaffold/generators/components.py [hunk 2]
```diff
@@ -70,11 +70,25 @@ def get_file_content(self) -> str:
...
         ]
 
         return "\n".join(file_content)
 
+    def _get_pre_condition_rounds_str(self) -> str:
+        """Get string with pre_conditions rounds."""
+        return _indent_wrapper("\n".join([f"\t{round}: []," for round in self.rounds]))
```
Comment (Adamantios) [inline autonomy/fsm/scaffold/generators/components.py:84]: We only need to do this for the initial rounds.

File: autonomy/fsm/scaffold/generators/components.py [hunk 3]
```diff
@@ -70,11 +70,25 @@ def get_file_content(self) -> str:
...
+
+    def _get_post_condition_rounds_str(self) -> str:
+        """Get string with pre_conditions rounds."""
+        return _indent_wrapper(
+            "\n".join([f"\t{round}: []," for round in self.degenerate_rounds])
```
Comment (Adamantios) [inline autonomy/fsm/scaffold/generators/components.py:89]: We need this for the terminal rounds instead.

File: autonomy/fsm/scaffold/generators/components.py [hunk 4]
```diff
@@ -24,7 +24,7 @@
 
 from aea.protocols.generator.common import _camel_case_to_snake_case
 
-from autonomy.fsm.scaffold.base import AbstractFileGenerator
+from autonomy.fsm.scaffold.base import AbstractFileGenerator, _indent_wrapper
```
Comment (Adamantios) [inline autonomy/fsm/scaffold/generators/components.py:27]: No need to import this if you use the `template_kwargs` in the base instead.

File: autonomy/fsm/scaffold/generators/components.py (hunk 3 above)
Comment (solarw) [reply autonomy/fsm/scaffold/generators/components.py:89]: done

File: autonomy/fsm/scaffold/generators/components.py (hunk 1 above)
Comment (solarw) [reply autonomy/fsm/scaffold/generators/components.py:77]: done

File: autonomy/fsm/scaffold/generators/components.py (hunk 4 above)
Comment (solarw) [reply autonomy/fsm/scaffold/generators/components.py:27]: done

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**